judge/views/api/api_v1.py

```diff
@@ -114,13 +114,30 @@ def api_v1_user_info(request, user):
     profile = get_object_or_404(Profile, user__username=user)
     submissions = list(Submission.objects.filter(case_points=F('case_total'), user=profile, problem__is_public=True, problem__is_organization_private=False)
                        .values('problem').distinct().values_list('problem__code', flat=True))
-    return JsonResponse({
+    resp = {
         'display_name': profile.name,
         'points': profile.points,
         'rank': profile.display_rank,
         'solved_problems': submissions,
         'organizations': list(profile.organizations.values_list('key', flat=True)),
-    })
+    }
+
+    last_rating = profile.ratings.order_by('-contest__end_time').first()
```

Comment:
Can you just reuse the query you are using below? Otherwise, at least get rid of the sort on that query.